src/geometry/euclidean/monge_point.lean

```diff
@@ -460,6 +546,76 @@ begin
+a vertex that was not replaced is the corresponding side of `t₁`. -/
+lemma altitude_replace_orthocenter_eq_affine_span {t₁ t₂ : triangle ℝ P} {i₁ i₂ i₃ j₁ j₂ j₃ : fin 3}
+    (hi₁₂ : i₁ ≠ i₂) (hi₁₃ : i₁ ≠ i₃) (hi₂₃ : i₂ ≠ i₃) (hj₁₂ : j₁ ≠ j₂) (hj₁₃ : j₁ ≠ j₃)
+    (hj₂₃ : j₂ ≠ j₃) (h₁ : t₂.points j₁ = t₁.orthocenter) (h₂ : t₂.points j₂ = t₁.points i₂)
+    (h₃ : t₂.points j₃ = t₁.points i₃) :
+  t₂.altitude j₂ = affine_span ℝ {t₁.points i₁, t₁.points i₂} :=
+begin
+  symmetry,
+  rw [←h₂, t₂.affine_span_insert_singleton_eq_altitude_iff],
+  rw [h₂],
+  use (injective_of_affine_independent t₁.independent).ne hi₁₂,
+  have he : affine_span ℝ (set.range t₂.points) = affine_span ℝ (set.range t₁.points),
+  { refine ext_of_direction_eq _
+      ⟨t₁.points i₃, mem_affine_span ℝ ⟨j₃, h₃⟩, mem_affine_span ℝ (set.mem_range_self _)⟩,
+    refine eq_of_le_of_findim_eq (direction_le (span_points_subset_coe_of_subset_coe _)) _,
+    { have hu : (finset.univ : finset (fin 3)) = {j₁, j₂, j₃}, { clear h₁ h₂ h₃, dec_trivial! },
+      rw [←set.image_univ, ←finset.coe_univ, hu, finset.coe_insert, finset.coe_insert,
+          finset.coe_singleton, set.image_insert_eq, set.image_insert_eq, set.image_singleton,
+          h₁, h₂, h₃, set.insert_subset, set.insert_subset, set.singleton_subset_iff],
+      exact ⟨t₁.orthocenter_mem_affine_span,
+             mem_affine_span ℝ (set.mem_range_self _),
+             mem_affine_span ℝ (set.mem_range_self _)⟩ },
+    { rw [direction_affine_span, direction_affine_span,
+          findim_vector_span_of_affine_independent t₁.independent (fintype.card_fin _),
+          findim_vector_span_of_affine_independent t₂.independent (fintype.card_fin _)] } },
+  rw he,
+  use mem_affine_span ℝ (set.mem_range_self _),
+  have hu : finset.univ.erase j₂ = {j₁, j₃}, { clear h₁ h₂ h₃, dec_trivial! },
+  rw [hu, finset.coe_insert, finset.coe_singleton, set.image_insert_eq, set.image_singleton,
+      h₁, h₃],
+  have hle : (t₁.altitude i₃).direction.orthogonal ≤
+    (affine_span ℝ ({t₁.orthocenter, t₁.points i₃} : set P)).direction.orthogonal :=
+      submodule.orthogonal_le (direction_le (affine_span_orthocenter_point_le_altitude _ _)),
+  refine hle ((t₁.vector_span_le_altitude_direction_orthogonal i₃) _),
+  have hui : finset.univ.erase i₃ = {i₁, i₂}, { clear hle h₂ h₃, dec_trivial! },
+  rw [hui, finset.coe_insert, finset.coe_singleton, set.image_insert_eq, set.image_singleton],
+  refine vsub_mem_vector_span ℝ (set.mem_insert _ _)
+    (set.mem_insert_of_mem _ (set.mem_singleton _))
+end
+
+/-- Suppose we are given a triangle `t₁`, and replace one of its
+vertices by its orthocenter, yielding triangle `t₂` (with vertices not
+necessarily listed in the same order).  Then the orthocenter of `t₂`
+is the vertex of `t₁` that was replaced. -/
+lemma orthocenter_replace_orthocenter_eq_point {t₁ t₂ : triangle ℝ P} {i₁ i₂ i₃ j₁ j₂ j₃ : fin 3}
+    (hi₁₂ : i₁ ≠ i₂) (hi₁₃ : i₁ ≠ i₃) (hi₂₃ : i₂ ≠ i₃) (hj₁₂ : j₁ ≠ j₂) (hj₁₃ : j₁ ≠ j₃)
```

Comment:
I wonder if there's a bundled way to state `(hi₁₂ : i₁ ≠ i₂) (hi₁₃ : i₁ ≠ i₃) (hi₂₃ : i₂ ≠ i₃)`...

Reply:
Well, it's possible to write things such as `(hi : function.injective ![i₁, i₂, i₃])`, but I think for most cases of concrete choices of indices it will be more convenient to produce the three `≠` hypotheses. (The general principle that can be seen at work in these lemmas is that the manipulations involved in proving something for general `n` tend to be more convenient than those involved in proving something specific to the case of `3`.)

Reply:
Using `list.pairwise` you can say `[i₁, i₂, i₃].pairwise (≠)`

Reply:
If what you have when using this sort of result is a list or a function from `fin 3` or anything like that, in which all three indices play exactly the same role, the bundled forms make sense. But I don't think that's the main use case for such lemmas (indeed, in this lemma, the indices don't all play the same role; one is the vertex being replaced, the other two are the vertices not being replaced). When the indices don't all play the same role, that's liable to mean that inequality of different indices may be derived in different ways (possibly from `norm_num`, when the indices are 0, 1 and 2, possibly from `dec_trivial`, when some of them are arbitrary `fin 3` values different from one or two such values passed into a lemma using this one, for example), making three separate hypotheses more convenient to pass to this lemma than one bundled hypothesis.

Reply:
Note that @shingtaklam1324's version is straightforward to produce with
```
import data.list.pairwise

example {α : Type} {i₁ i₂ i₃ : α} (hi₁₂ : i₁ ≠ i₂) (hi₁₃ : i₁ ≠ i₃) (hi₂₃ : i₂ ≠ i₃) : [i₁, i₂, i₃].pairwise (≠) := by tidy; exact list.pairwise.nil
```
It looks to me like `pairwise.nil` is missing a simp lemma.

Reply:
With #4281, the proof above is just `by simp` - so in principle a caller of `orthocenter_replace_orthocenter_eq_point ` could just use `refine orthocenter_replace_orthocenter_eq_point _ h₁ h₂ h₃, simp,` and they'd end up at the three hypotheses you list here. Up to you whether you want to explore this further.

Reply:
I tend to think that alternative options for how to express that these indices are distinct would best be explored once we have users of `orthocenter_replace_orthocenter_eq_point`. (I have a definition and some basic properties of orthocentric systems, but as well as this PR those properties depend on some definitions / lemmas about collinearity that I want to rework before PRing them.)